lib/iris/tests/unit/data_manager/test_DataManager.py

```diff
@@ -0,0 +1,775 @@
+    def test(self):
+        dm = DataManager(np.array(0))
+        method = 'iris._data_manager.DataManager._deepcopy'
+        return_value = mock.sentinel.return_value
+        with mock.patch(method) as mocker:
+            mocker.return_value = return_value
+            result = copy.deepcopy(dm)
+            self.assertEqual(mocker.call_count, 1)
+            [args], kwargs = mocker.call_args
+            self.assertEqual(kwargs, dict())
+            self.assertEqual(len(args), 2)
+            expected = [return_value, [dm]]
+            for item in six.itervalues(args):
+                self.assertIn(item, expected)
+        self.assertIs(result, return_value)
+
+
+class Test___eq__(tests.IrisTest):
+    def setUp(self):
+        self.shape = (2, 3, 4)
+        self.size = np.prod(self.shape)
+        self.real_array = np.arange(self.size).reshape(self.shape)
+
+    def test_real_with_real(self):
+        dm1 = DataManager(self.real_array)
+        dm2 = DataManager(self.real_array.copy())
+        self.assertEqual(dm1, dm2)
+
+    def test_real_with_real_failure(self):
+        dm1 = DataManager(self.real_array)
+        dm2 = DataManager(np.ones(self.shape))
+        self.assertFalse(dm1 == dm2)
+
+    def test_real_with_real__dtype_failure(self):
+        dm1 = DataManager(self.real_array)
+        dm2 = DataManager(self.real_array * 1.0)
+        self.assertFalse(dm1 == dm2)
+
+    def test_real_with_lazy_failure(self):
+        dm1 = DataManager(self.real_array)
+        dm2 = DataManager(as_lazy_data(self.real_array))
+        self.assertFalse(dm1 == dm2)
+        self.assertFalse(dm2 == dm1)
+
+    def test_lazy_with_lazy(self):
+        dm1 = DataManager(as_lazy_data(self.real_array))
+        dm2 = DataManager(as_lazy_data(self.real_array))
+        self.assertEqual(dm1, dm2)
+
+    def test_lazy_with_lazy_failure(self):
+        dm1 = DataManager(as_lazy_data(self.real_array))
+        dm2 = DataManager(as_lazy_data(self.real_array) * 10)
+        self.assertFalse(dm1 == dm2)
+
+    def test_lazy_with_lazy__dtype_failure(self):
+        dm1 = DataManager(as_lazy_data(self.real_array))
+        dm2 = DataManager(as_lazy_data(self.real_array) * 1.0)
+        self.assertFalse(dm1 == dm2)
+
+    def test_lazy_with_lazy__realised_dtype(self):
+        dtype = np.dtype('int16')
+        dm1 = DataManager(as_lazy_data(self.real_array),
+                          realised_dtype=dtype)
+        dm2 = DataManager(as_lazy_data(self.real_array),
+                          realised_dtype=dtype)
+        self.assertEqual(dm1, dm2)
+
+    def test_lazy_with_lazy__realised_dtype_failure(self):
+        dm1 = DataManager(as_lazy_data(self.real_array),
+                          realised_dtype=np.dtype('int8'))
+        dm2 = DataManager(as_lazy_data(self.real_array),
+                          realised_dtype=np.dtype('int16'))
+        self.assertFalse(dm1 == dm2)
+
+    def test__non_DataManager_failure(self):
```

Comment:
why does this have 2 underscores, i.e. why is it not `test_non_DataManager_failure`